OpenRA.Mods.Common/Traits/Carryable.cs

```diff
@@ -103,33 +104,33 @@ public virtual bool Reserve(Actor self, Actor carrier)
 			Carrier = carrier;
 
 			if (reservedToken == Actor.InvalidConditionToken)
-				reservedToken = self.GrantCondition(Info.ReservedCondition);
+				reservedToken = Self.GrantCondition(Info.ReservedCondition);
 
 			return true;
 		}
 
-		public virtual void UnReserve(Actor self)
+		public virtual void UnReserve()
```

Comment:
I don't think this place needs an actor: as a trait it is attached to the very actor and should only take effect on this actor, and previous code has a possibility to make the user assign the wrong actor when coding.

Meanwhile, pass less parameters (especially unecessary one) can improve perf when call this function.

Reply:
And the same opion on other places.

Reply:
> I don't think this place needs an actor: as a trait it is attached to the very actor and should only take effect on this actor, and previous code has a possibility to make the user assign the wrong actor when coding.
> 
> Meanwhile, pass less parameters (especially unecessary one) can improve perf when call this function.

It's by OpenRA's convention. And specifically here it is necessary, you just made it fetch from slightly deeper memory instead

Reply:
> specifically here it is necessary, you just made it fetch from slightly deeper memory instead

The Actor para can only be actor itself, we cannot use other actor as para -- without proper documentation and check (`if self != this.self then throw an exception`) this will misguide new coder and cause bug, and it is not an interface here like `ITick` actually so I think the actor self can be removed.

For example, by using our convention, at "Carryall.cs", line 101 we got a `self` on actor itself, but 
 at line 237 and 250 we also use a para `self` in `UnreserveCarryable`, this is confusing and we don't even has a check on `if self != this.self then throw an exception`, I can hardly say this is a proper convention, idk what do you guys think.

Reply:
the idea is that the engine is way too large to change the convention, it would take multiple full work days if not a week of work just to do the changes, and then there's also reviewing and testing for regressions. And we are not even sure if it will have a positive performance impact in the end. It's best to just follow the convention

Reply:
> the idea is that the engine is way too large to change the convention

We don't have to change the overall convention IMO, we just need to change carryall related, because only carryall and carryable here has the problem of having a private `self` and a para `self`, which is confusing in the overall convention.

> And we are not even sure if it will have a positive performance impact in the end

Normally in coding, less para to pass means less var copying, even reference (as a var storage a pointer),